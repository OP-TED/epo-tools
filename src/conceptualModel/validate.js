import { hasPrefix } from '../prefix/prefix.js'
import { ATTRIBUTE, INHERITANCE } from './const.js'

function validateNode (node) {
  return hasPrefix(node.name) ? [] : [
    {
      severity: 'error', desc: `no prefix for ${node.name}`,
    }]
}

function validateEdge (edge) {
  const {
    type, source, predicate, target, quantifiers: { quantifiersDeclared },
  } = edge
  const result = []

  const requiresTargetPrefix = type !== ATTRIBUTE
  const requiresPredicate = type !== INHERITANCE
  const requiresQuantifiers = type !== INHERITANCE

  if (!quantifiersDeclared && requiresQuantifiers) {
    result.push({
      severity: 'warning', desc: `No quantifiers defined`,
    })
  }
  if (!source) {
    result.push({
      severity: 'error', desc: `No source defined`,
    })
  } else if (!hasPrefix(source)) {
    result.push({
      severity: 'error', desc: `No prefix for source [${source}]`,
    })
  }

  if (requiresPredicate) {
    if (!predicate) {
      result.push({
        severity: 'error', desc: `No predicate defined`,
      })
    } else if (!hasPrefix(predicate)) {
      result.push({
        severity: 'error', desc: `No prefix for predicate [${predicate}]`,
      })
    }
  }

  if (!target) {
    result.push({
      severity: 'error', desc: `No target defined`,
    })
  } else if (requiresTargetPrefix && !hasPrefix(target)) {
    result.push({
      severity: 'error', desc: `No prefix for target [${target}]`,
    })
  }

  return result
}

export { validateNode, validateEdge }
