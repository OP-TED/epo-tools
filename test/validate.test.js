import { expect } from 'expect'
import toMatchSnapshot from 'expect-mocha-snapshot'
import { describe, it } from 'mocha'
import { validateEdge, validateNode } from '../src/conceptualModel/validate.js'
import { UNDER_REVIEW } from '../src/config.js'
import { getEpoJson } from '../src/epo/readEpo.js'

expect.extend({ toMatchSnapshot })

describe('validate', () => {

  const { edges, nodes } = getEpoJson(UNDER_REVIEW)

  it(`validateNode`, function () {
    const result = nodes.map(validateNode).filter(x => x.length)
    expect(result).toMatchSnapshot(this)
  })

  it(`validateEdge`, function () {
    const result = edges.map(validateEdge).filter(x => x.length)
    expect(result).toMatchSnapshot(this)
  })
})

