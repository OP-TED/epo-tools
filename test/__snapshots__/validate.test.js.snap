// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate validateEdge 1`] = `
Array [
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pmc]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pmc]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [can-modif]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [compl]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pin-buyer]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pin-only]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form6]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form13]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pin-rtl]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [compl]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pin-cfc-standard]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [pin-cfc-social]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [qu-sy]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cn-standard]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [subco]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cn-desg]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cn-social]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form1]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [(EU) 2019/1780]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [can-standard]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [can-social]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [can-desg]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form2]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form3]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form8]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form12]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form15]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form20]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form21]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [planning]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [dir-awa-pre]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form1]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form4]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form4]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form1]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form5]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form7]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form8]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form12]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form13]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form15]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form20]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form22]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [competition]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cont-modif]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cont-modif]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [cont-modif]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form15]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form20]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form16]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form15]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form8]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form17]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form19]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [(EU) 2019/1780]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [(EU) 2015/1986]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [(EU) 2015/1986]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [dir-awa-pre]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [result]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [dir-awa-pre]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [subco]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [can-desg]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [can-modif]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [can-social]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [can-standard]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [veat]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [cn-desg]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [cn-social]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [cn-standard]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pin-cfc-standard]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [Directive 81]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [veat]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [dir-awa-pre]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [dir-awa-pre]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [result]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pin-buyer]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pin-cfc-social]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pin-rtl]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [qu-sy]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [pin-only]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [Directive 23]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [Directive 24]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [Directive 25]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
    Object {
      "desc": "No prefix for source [cont-modif]",
      "severity": "error",
    },
    Object {
      "desc": "No predicate defined",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [planning]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [competition]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form22]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form22]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form22]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 25]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 23]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Directive 81]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form18]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form24]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No prefix for target [Form4]",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
  ],
  Array [
    Object {
      "desc": "No quantifiers defined",
      "severity": "warning",
    },
  ],
]
`;

exports[`validate validateNode 1`] = `
Array [
  Array [
    Object {
      "desc": "no prefix for pmc",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for planning",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for competition",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for dir-awa-pre",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for result",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for cont-modif",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for compl",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for can-standard",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for veat",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for cn-desg",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for cn-social",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for cn-standard",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for pin-cfc-standard",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for pin-buyer",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for pin-rtl",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for pin-only",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for qu-sy",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Directive 23",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Directive 24",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Directive 25",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Directive 81",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for (EU) 2019/1780",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for subco",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for can-desg",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for can-modif",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for can-social",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form1",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form2",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form3",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form4",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form16",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form5",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form6",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form7",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form8",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form12",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form13",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form15",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for pin-cfc-social",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form17",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form18",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form19",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form20",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form21",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form22",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form23",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form24",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for Form25",
      "severity": "error",
    },
  ],
  Array [
    Object {
      "desc": "no prefix for (EU) 2015/1986",
      "severity": "error",
    },
  ],
]
`;
